test-alerts.cpp

```diff
@@ -4,5 +4,17 @@
 #include "typewise-alert.h"
 
 TEST_CASE("infers the breach according to limits") {
-  REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+	BMS_Alerter alerter;
+	EmailLogger email;
+	BatteryCharacter battery_char;
+	battery_char.coolingType = PASSIVE_COOLING;
+	REQUIRE(alerter.checkAndAlert(email, battery_char, 40.0) == TOO_HIGH);
+	REQUIRE(alerter.checkAndAlert(email, battery_char, -1.0) == TOO_LOW);
```

Comment:
These REQUIRES state that checkAndAlert converts from the passed parameters to one HIGH/LOW return value.
If you had to restate its functionality in terms of data-conversions, what would you state?